### ISSUE TITLE
Build flavor flag interfering with no-sound-null-safety flag on IOS

### ISSUE THREAD

**Issue by ghensto** [flutter/flutter#79728]
``` 
[✓] Flutter (Channel beta, 2.1.0-12.2.pre, on macOS 11.3 20E5224a darwin-x64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.55.0)
[✓] Connected device (5 available)

 No issues found! 
```

**When I run:** `flutter run --no-sound-null-safety --flavor dev -t lib/main_dev.dart`
I get
 ``` Xcode's output:
↳
    Error: Cannot run with sound null safety, because the following dependencies
    don't support null safety:

     - package:flutter_voip_push_notification

    For solutions, see https://dart.dev/go/unsound-null-safety

    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete

Could not build the precompiled application for the device.

 ```

**When I run:** `flutter run --no-sound-null-safety -t lib/main_dev.dart`, it works fine

**Comment by dafinoer** [comment]
@ghensto 
your dependecies need to migrate null safety.

**Comment by TahaTesser** [comment]
Hi @ghensto 
I see there's an open issue addressing the case you described.
Please follow up on that issue, Closing as duplicate of https://github.com/flutter/flutter/issues/77912
If you disagree, please write in the comments and I will reopen it.
Thank you
